props/saber_BC_buttons.h

```diff
@@ -477,738 +1492,1266 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
+#if defined(ENABLE_AUDIO) && defined(AUDIO_CLASH_SUPPRESSION_LEVEL)
+          + dynamic_mixer.audio_volume() * (AUDIO_CLASH_SUPPRESSION_LEVEL * 1E-10) * dynamic_mixer.get_volume()
+#endif
+         ))) {
+      if ( (accel_ - fusor.down()).len2() > (accel - fusor.down()).len2() ) {
+        diff = -diff;
+      }
+
+      bool stab = (fabs(diff.x) > 2.0 * sqrtf(diff.y * diff.y + diff.z * diff.z)) &&
+            fusor.swing_speed() < 150;
+
+      if (!clash_pending1_) {
+        forward_stab_ = (stab && (diff.x < 0));
+        if (stab) PVLOG_DEBUG << "**** DoAccel >>>>>>>> - forward_stab_ = " << (forward_stab_ = (diff.x < 0)) << "\n";
+        clash_pending1_ = true;
+        pending_clash_is_stab1_ = stab;
+        pending_clash_strength1_ = v;
+      } else {
+        pending_clash_strength1_ = std::max<float>(v, (float)pending_clash_strength1_);
+      }
+    }
+    accel_ = accel;
+  }
+
+  // Handle lockup or drag for the main blade when only the main blade is on
+  void MainBladeLockupOrDrag() {
+    if (fusor.angle1() < -M_PI / 4) {
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_MAIN_BLADE_SET);
+    } else if (!battle_mode_) {
+      SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL, BC_MAIN_BLADE_SET);
+    }
+  }
+
+  // Handle lockup or drag for the second blade when only the second blade is on
+  void SecondBladeLockupOrDrag() {
+    if (fusor.angle1() > M_PI / 4) {
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_SECOND_BLADE_SET);
+    } else if (!battle_mode_) {
+     SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL, BC_SECOND_BLADE_SET);
+    }
+  }
+
+  // Handle lockup or drag for dual blades when both blades are on
+  void DualBladesLockupOrDrag() {
+    if (fusor.angle1() < -M_PI / 4) {
+      // Main blade is pointing down, and within drag range
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_MAIN_BLADE_SET);
+    } else if (fusor.angle1() > M_PI / 4) {
+      // Main blade is pointing up, and within drag range for the second blade
+      SaberBase::SetLockup(SaberBase::LOCKUP_DRAG, BC_SECOND_BLADE_SET);
+    } else {
+      // Otherwise, apply normal lockup based on which blade is above the horizon
+      SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL, GetBladeAboveHorizon());
+    }
+  }
+
+#endif  // BC_DUAL_BLADES
+
+  RefPtr<BufferedWavPlayer> wav_player;
+
+  bool Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+#ifdef BC_DUAL_BLADES
+    GetThrustBladeLocation();
 #endif
 
-// Spoken Battery Level in volts
-    case EVENTID(BUTTON_POWER, EVENT_THIRD_SAVED_CLICK_SHORT, MODE_OFF):
-      if (!mode_volume_) {
-        // Avoid weird battery readings when using USB
-        if (battery_monitor.battery() < 0.5) {
-          sound_library_.SayTheBatteryLevelIs();
-          sound_library_.SayDisabled();
+    if (event == EVENT_TWIST) {
+      PVLOG_DEBUG << "**** Detected EVENT_TWIST in Event2, stopping timer and resetting saved twist.\n";
+      saved_twist_ = 0;
```

Comment:
I don't believe so.
If we use the saved twist in DoSavedTwist(), then it's reset.
But also, if we disregard the saved twist because a normal twist occurred in Event2, it also needs to be reset so that when CheckSavedTwist() is called, it's not lingering... am I wrong?

Reply:
Right, this is a *normal* twist event.
I was mixing up the events.
Not sure if resetting this here will really fix how twist events interact or not though....